webpack-utils.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ *
+ * @fileoverview Extra utilities used by webpack
+ */
+
+var PageMeta = function(name, description) {
+  this.name = name;
+  this.description = description;
+  this.referrer = 'no-referrer';
+  this.viewport = 'width=device-width, initial-scale=1.0, user-scalable=yes';
+
+  this['application-name'] = '<[ siteName ]>';
+  this['msapplication-square310x310logo'] = (
+    '<[ getAssetUrl(\'/assets/images/logo/msapplication-large.png\') ]>');
+  this['msapplication-wide310x150logo'] = (
+    '<[ getAssetUrl(\'/assets/images/logo/msapplication-wide.png\') ]>');
+  this['msapplication-square150x150logo'] = (
+    '<[ getAssetUrl(\'/assets/images/logo/msapplication-square.png\') ]>');
+  this['msapplication-square70x70logo'] = (
+    '<[ getAssetUrl(\'/assets/images/logo/msapplication-tiny.png\') ]>');
+
+  this['og:type'] = {
+    property: 'og:type',
+    content: 'article'
+  };
+  this['og:site_name'] = {
+    property: 'og:site_name',
+    content: 'Oppia'
+  };
+  this['og:url'] = {
+    property: 'og:url',
+    content: '<[pageUrl]>'
+  };
+  this['og:title'] = {
+    property: 'og:title',
+    content: name
+  };
+  this['og:description'] = {
+    property: 'og:description',
+    content: description
+  };
+  this['og:image'] = {
+    property: 'og:image',
+    content: '<[ getAssetUrl(\'/assets/images/logo/288x288_logo_mint.png\') ]>'
+  };
+};
+
+PageMeta.prototype.toHTML = function() {
+  var html = '<meta charset="UTF-8">\n';
+  for (var meta in this) {
+    if (this.hasOwnProperty(meta)) {
+      if (typeof this[meta] === 'object') {
+        html += '<meta ' + Object.keys(this[meta])[0] + '="' +
+        Object.values(this[meta])[0] + '" ' + Object.keys(this[meta])[1] +
+        '="' + Object.values(this[meta])[1] + '">\n';
+      } else {
+        html += '<meta name="' + meta + '" content="' + this[meta] + '">\n';
+      }
+    }
+  }
+  return html;
+};
+
+var utilities = {
+  getMetas: function() {
+    var getStarted = new PageMeta(
```

Comment:
I'm a bit confused by this syntax. The return value is an object, so does that mean "new PageMeta(...)" is a key value pair?

Reply:
Yes.
`new PageMeta(...)` returns an object which is most often a key-value pair. Except, in this case, we are also adding a method `toHTML()` to the object

Reply:
I don't think I understand ... how can one return a key-value pair? I've never heard of that as being an actual "typed" thing.

Reply:
I don't think I totally understand what you mean by "actual typed thing" but using `new` before a function in Javascript makes it a constructor function and it returns `this` explicitly.

Reply:
I imagined that "new PageMeta(...)" returns an instance of a PageMeta object. But then you are returning it in

   { ... } 

which is typically the syntax for a dict with key-value pairs.

So, either a PageMeta object is a "key-value pair" (which doesn't make sense to me -- it's not a "type" like number, string, etc.) or I'm misunderstanding something in JS. Would you mind clarifying? Sorry for my ignorance, it's possible that this is a different construct from what I'm thinking.

Reply:
I imagined that "new PageMeta(...)" returns an instance of a PageMeta object. But then you are returning it in

   { ... } 

which is typically the syntax for a dict with key-value pairs.

So, either a PageMeta object is a "key-value pair" (which doesn't make sense to me -- since "key-value pair" is not a "type" like number, string, etc.) or I'm misunderstanding something in JS. Would you mind clarifying? Sorry for my ignorance, it's possible that this is a different construct from what I'm thinking.

Reply:
Yes.
Here is what is happening there.
The new PageMeta constructs the object and here, it is saved in the `getStarted` variable.
We need to access this object in the webpack.config.ts so I am returning it in the { ... }.
Returning `{ getStarted }` there is similar to saying `{ getStarted: getStarted }`
Since it is the same name, it is allowed in JS to just type `{ getStarted }`
I hope this clarifies it.

Reply:
The object is returned that way so that other pages say splash can be done in a similar manner.
`var splash = new PageMeta(...)`
Then added to the object being returned as so
`{getStarted, splash}`

Reply:
With new changes, this file has been removed